compiler/rustc_mir_transform/src/ssa.rs

```diff
@@ -187,15 +187,9 @@ impl SsaLocals {
     }
 }
 
-#[derive(Copy, Clone, Debug, PartialEq, Eq)]
-enum LocationExtended {
-    Plain(Location),
-    Arg,
```

Comment:
https://github.com/rust-lang/rust/pull/113915 adds a variant to this enum to represent an assignment along an edge. That's useful for call and yield terminators. Can this be modelled with `DefLocation`, or will I have to revert this?

Reply:
Thinking a bit more about this, reusing `DefLocation` may be the right solution. `rustc_codegen_ssa` only relies on it to compute domination. The difference is the semantics we give to call and yield terminators.

#113915 puts the assignment along the edge. `rustc_codegen_ssa` puts the assignment *before* the edge. They are equivalent iff using the local's value on the unwind edge is UB. Equivalently, that would mean assigning `undef` on the unwind edge.

I guess this is a question for @rust-lang/opsem?

Reply:
So concretely the question is, what is guaranteed about the contents of the return place when the function unwinds?

That's a good question I hadn't considered yet. Miri will de-initialize its content before calling the function, but anything the callee writes to the return place before unwinding is guaranteed to remain there.

Reply:
I agree. The better solution is to fix the codegen implementation and then share the result.

Reply:
On the semantic question, I realized what I said above was wrong, the caller-provided return place will always be fully uninitialized in Miri upon unwinding. I opened https://github.com/rust-lang/unsafe-code-guidelines/issues/468 to potentially discuss this.

I don't understand what your conclusion means for the semantics, but this should give optimizations and codegen maximal freedom.